libavcodec/vaapi_encode.c

```diff
@@ -2520,6 +2550,13 @@ av_cold int ff_vaapi_encode_init(AVCodecContext *avctx)
         }
     }
 
+    ctx->encode_fifo = av_fifo_alloc((ctx->async_depth > MAX_PICTURE_REFERENCES ?
```

Comment:
1. If we have this, How to handle low delay encode? 
2. could we adjust async_depth in one place?